api/v1alpha1/quaysecretengineconfig_webhook.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"errors"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// log is for logging in this package.
+var quaysecretengineconfiglog = logf.Log.WithName("quaysecretengineconfig-resource")
+
+func (r *QuaySecretEngineConfig) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+
+//+kubebuilder:webhook:path=/mutate-redhatcop-redhat-io-v1alpha1-quaysecretengineconfig,mutating=true,failurePolicy=fail,sideEffects=None,groups=redhatcop.redhat.io,resources=quaysecretengineconfigs,verbs=create;update,versions=v1alpha1,name=mquaysecretengineconfig.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Defaulter = &QuaySecretEngineConfig{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *QuaySecretEngineConfig) Default() {
```

Comment:
if you are not setting any default, are you sure you need this?

Reply:
I see some Secrets engines setting finalizers, while others do not. do you have any advice around this topic?

Reply:
for some secret engine the config object cannot be delete as it's 1 to 1 with the engine mount. so we have decided not to do anything on the finalize. but you should have finalizes if this engine support an 1 to many relation. It should ve definitely needed for role as they are always one to many.

Reply:
You should be able to define multiple configs at different paths. so i can look to add in support for the finalizer and will update

Reply:
Not sure if the plugin falls into that or not. i believe we have a 1:1 with the path (ie you cant have multiple configs on the same path)